syft/pkg/cataloger/python/license_test.go

```diff
@@ -0,0 +1,175 @@
+package python
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFormatPyPiRegistryURL(t *testing.T) {
+	tests := []struct {
+		name          string
+		version       string
+		expected      string
+		expectedError error
+	}{
+		{
+			name:          "package1",
+			version:       "1.0",
+			expected:      "https://pypi.org/pypi/package1/1.0/json",
+			expectedError: nil,
+		},
+		{
+			name:          "package-1",
+			version:       "",
+			expected:      "https://pypi.org/pypi/package-1/json",
+			expectedError: nil,
+		},
+		{
+			name:          "_",
+			version:       "a",
+			expected:      "https://pypi.org/pypi/_/a/json",
+			expectedError: nil,
+		},
+		{
+			name:          "",
+			version:       "a",
+			expected:      "",
+			expectedError: fmt.Errorf("unable to format pypi request for a blank package name"),
+		},
+	}
+
+	cfg := DefaultCatalogerConfig()
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := formatPypiRegistryURL(cfg.PypiBaseURL, test.name, test.version)
+
+			require.Equal(t, test.expected, got)
+			require.Equal(t, test.expectedError, err)
+		})
+	}
+
+}
+
+func TestGetLicenseFromPypiRegistry(t *testing.T) {
+	mux, url, teardown := setupPypiRegistry()
+	defer teardown()
+
+	tests := []struct {
+		name            string
+		version         string
+		requestHandlers []handlerPath
+		expected        string
+		expectedError   error
+	}{
+		{
+			name:    "certifi",
+			version: "2025.10.5",
+			requestHandlers: []handlerPath{
+				{
+					path:    "/certifi/2025.10.5/json",
+					handler: generateMockPypiRegistryHandler("test-fixtures/pypi-remote/registry_response.json"),
+				},
+			},
+			expected: "MPL-2.0",
+		},
+		{
+			name:    "package",
+			version: "1.0",
+			requestHandlers: []handlerPath{
+				{
+					path:    "/package/1.0/json",
+					handler: generateMockPypiRegistryHandlerWithStatus("", http.StatusNotFound),
+				},
+			},
+			expected:      "",
+			expectedError: fmt.Errorf("unable to get package from pypi registry"),
+		},
+		{
+			name:    "package",
+			version: "2.0",
+			requestHandlers: []handlerPath{
+				{
+					path:    "/package/2.0/json",
+					handler: generateMockPypiRegistryHandler("test-fixtures/pypi-remote/registry_response_bad.json"),
+				},
+			},
+			expected:      "",
+			expectedError: fmt.Errorf("unable to parse license from pypi registry: EOF"),
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// set up the mock server
+			for _, handler := range tc.requestHandlers {
+				mux.HandleFunc(handler.path, handler.handler)
+			}
+			got, err := getLicenseFromPypiRegistry(url, tc.name, tc.version)
+			require.Equal(t, tc.expected, got)
+			if err != nil {
+				require.Equal(t, tc.expectedError.Error(), err.Error())
+			} else {
+				require.Equal(t, tc.expectedError, err)
```

Comment:
tc.expectedError could be nil here. there are some require helpers for errors like `require.NoError` and `require.ErrorContains`. I think this could be a bit better written like:
```
if tc.expectedError != nil {
  require.ErrorContains(t, err, tc.expectedError.Error())
} else {
  require.NoError(t, err)
```
or similar